### PR TITLE
[Tune] Ray Client support for examples & CI

### PR DESCRIPTION
**Pull request by amogkam**
Out of the box Ray Client support for Tune examples. Also add key examples with Ray Client to run in CI

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -258,6 +258,14 @@
     - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=py37,-flaky python/ray/tune/...
     - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only python/ray/util/xgboost/...
 
+- label: ":tropical_fish: ML Libraries w/ Ray Client Examples."
```
Comment (richardliaw) [inline .buildkite/pipeline.yml:261]: tropical fish...

Comment (amogkam) [reply .buildkite/pipeline.yml:261]: beta clearance -> betafish -> tropical fish

File: python/ray/tune/examples/cifar10_pytorch.py [hunk 2]
```diff
@@ -206,21 +222,14 @@ def main(num_samples=10, max_num_epochs=10, gpus_per_trial=2):
...
-            best_trained_model = nn.DataParallel(best_trained_model)
-    best_trained_model.to(device)
-
-    checkpoint_path = os.path.join(best_trial.checkpoint.value, "checkpoint")
+    if args.server_address:
+        # If using Ray Client, we want to make sure checkpoint access
+        # happens on the server. So we wrap `test_best_model` in a Ray task.
+        ray.get(ray.remote(test_best_model).remote())
+    else:
+        test_best_model()
```
Comment (richardliaw) [inline python/ray/tune/examples/cifar10_pytorch.py:230]: hmm this example is user-facing (is used in docs). but this method is not shown -- do you need to revise the docs (or define the function within the __main_end__ scope?

File: python/ray/tune/tests/test_client.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    assert ray.util.client.ray.is_connected()
+    from ray.tune.examples.horovod_simple import tune_horovod
+    tune_horovod(
+        hosts_per_trial=1, slots_per_host=2, num_samples=2, use_gpu=False)
+
+
+def test_xgboost_example(start_client_server):
+    assert ray.util.client.ray.is_connected()
+    from ray.tune.examples.xgboost_example import tune_xgboost
+    tune_xgboost()
```
Comment (richardliaw) [inline python/ray/tune/tests/test_client.py:72]: I think you need a __main__ below here

**Review by richardliaw — COMMENTED** [review]
looks good! just one comment

File: python/ray/tune/examples/cifar10_pytorch.py (hunk 2 above)
Comment (amogkam) [reply python/ray/tune/examples/cifar10_pytorch.py:229]: Ah good point, updated

**Review by richardliaw — APPROVED** [review]

File: python/ray/tune/BUILD [hunk 4]
```diff
@@ -33,6 +33,14 @@ py_test(
     deps = [":tune_lib"],
 )
 
+py_test(
+    name = "test_client",
+    size = "large",
+    srcs = ["tests/test_client/py"],
```
Comment (amogkam) [inline python/ray/tune/BUILD:39]:
```suggestion
    srcs = ["tests/test_client.py"],
```

**Comment by amogkam** [comment]
Failing tests are all known failing tests & are unrelated. Going to merge this in.

**Pull request merged**